Sources/Apodini/Properties/PathParameter.swift

```diff
@@ -19,10 +22,21 @@ public struct PathParameter<Element: Codable & LosslessStringConvertible> {
     
     /// Accessing the projected value allows you to pass the `@PathParameter` to a `Handler` or `Component`
     public var projectedValue: Parameter<Element> {
-        Parameter(from: id)
+        Parameter(from: id, identifying: identifyingType)
     }
     
     
     /// Creates a new `@PathParameter`
-    public init() { }
+    public init() {
+        precondition(!isOptional(Element.self), "A `PathParameter` cannot annotate a property with Optional type!")
+    }
+
+    /// Creates a new `@PathParameter` specifically stating the type it identifies.
+    /// The identified type must conform to `Identifiable` and the property type of the `PathParameter`
+    /// must match the type of the `id` property of the  identified type.
+    ///
+    /// - Parameter type: The type the PathParameter value identifies.
+    public init<Type: Encodable & Identifiable>(identifying type: Type.Type = Type.self) where Element == Type.ID {
+        self.identifyingType = IdentifyingType(type: type, idType: type.ID.self)
```

Comment:
Added a call to `self.init()` running the same check for all initializers.
Anyways this would also have been caught in the `Parameter` precondition check.
But fixed with upcoming commit.

Reply:
Resolved in f82004a